crates/ol/mempool/src/ordering.rs

```diff
@@ -0,0 +1,145 @@
+//! Transaction ordering strategies for the mempool.
+
+use crate::types::{MempoolEntry, OLMempoolTxPayload};
+
+/// Trait for ordering transactions in the mempool.
+///
+/// Different strategies can be implemented (FIFO, fee-based, account-based, etc.)
+/// to determine transaction priority.
+pub(crate) trait OrderingStrategy: Send + Sync + 'static {
+    /// Compute the priority value for a transaction.
+    ///
+    /// Lower values = higher priority (will be selected first).
+    /// The priority is used to order transactions when retrieving them.
+    ///
+    /// Priority is a u128 value encoding both primary sort key (seq_no or slot)
+    /// and insertion_id for deterministic tie-breaking.
+    fn compute_priority(&self, entry: &MempoolEntry) -> u128;
+}
+
+/// FIFO (First-In-First-Out) ordering strategy.
+///
+/// For SnarkAccountUpdate transactions: Orders by sequence number (lower seq_no = higher priority).
```

Comment:
This is sub-ordering, we still need fifo for txs between different snark accounts